test/Riok.Mapperly.Tests/Mapping/UseStaticMapperTest.cs

```diff
@@ -359,4 +361,94 @@ public partial class Mapper
                 """
             );
     }
+
+    [Theory]
+    [InlineData(true)] // Usually when your IDE runs the source generator (references are other syntax trees)
+    [InlineData(false)] // When compiler produces final assembly (references are compiled assemblies)
+    public void UseStaticGenericMapperStaticMethodFromAnotherAssembly(bool asCompilationReference)
```

Comment:
I think the readability here would be improved if the common code is extracted in a method and two separate tests are created.